### PR TITLE
Make 'baseDomain' a required value

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it

There was a problem setting the `baseDomain` last week, and clusters kept being created relying on the default `baseDomain`, which was set to example.com. This made that the clusters were not being created successfully.

Instead, this value should be always be passed and fail if that's not the case.

### Checklist

- [X] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
-- Bump external-dns to latest release
```
Comment (zewolfe) [inline CHANGELOG.md:12]: Is this change log no longer needed?

Comment (zewolfe) [inline CHANGELOG.md:12]: Is this change log no longer needed?

Comment (fiunchinho) [reply CHANGELOG.md:12]: I think this was a mistake. We don't keep track of `external-dns` on this repo? :smile:

Comment (zewolfe) [reply CHANGELOG.md:12]: You're right 😄

**Review by zewolfe — APPROVED** [review]
👊🏾

**Pull request merged**